src/test_creation/modules/code_analyzer/analyzers/python.py

```diff
@@ -24,6 +25,14 @@ def read(self, file_path: str):
             self.content = f.read()
             self._tree = ast.parse(self.content)
 
+    @assert_have_read_content
+    def _get_function_lineno_map(self): # FIXME: when to use _xxx? when to use xxx?
```

Comment:
The decision of using underscore or not is based on more like a design choice rather than a rule.

Python does not have data hiding capabilities like some OO languages do, so technically everything in a class is accessible from the outside. However, by convention we denote "protected" properties/methods with *one* underscore in front for IDEs/parsers to not show suggestions in code completion, and denote "private" ones with *two* underscores in front. (Python will attempt to hide variables that start with two underscores, but they are still accessible if you know the way they hide it - refer to [this](https://docs.python.org/3/tutorial/classes.html#private-variables) and [this](https://docs.python.org/3/reference/expressions.html#atom-identifiers:~:text=Private%20name%20mangling,transformation%20is%20done.))